neo.UnitTests/Ledger/UT_MemoryPool.cs

```diff
@@ -253,5 +284,218 @@ public void TestInvalidateAll()
+            IEnumerable enumerable = _unit;
+            var enumerator = enumerable.GetEnumerator();
+            foreach (Transaction tx in _unit)
+            {
+                enumerator.MoveNext();
+                enumerator.Current.Should().BeSameAs(tx);
+            }
+        }
+
+        [TestMethod]
+        public void TestGetVerifiedTransactions()
+        {
+            var tx1 = CreateTransaction();
+            var tx2 = CreateTransaction();
+            _unit.TryAdd(tx1.Hash, tx1);
+            _unit.InvalidateVerifiedTransactions();
+            _unit.TryAdd(tx2.Hash, tx2);
+            IEnumerable<Transaction> enumerable = _unit.GetVerifiedTransactions();
+            enumerable.Count().Should().Be(1);
+            var enumerator = enumerable.GetEnumerator();
+            enumerator.MoveNext();
+            enumerator.Current.Should().BeSameAs(tx2);
+        }
+
+        [TestMethod]
+        public void TestReVerifyTopUnverifiedTransactionsIfNeeded()
+        {
+            NeoSystem TheNeoSystem = TestBlockchain.InitializeMockNeoSystem();
+            var s = Blockchain.Singleton.Height;
+            _unit = new MemoryPool(TheNeoSystem, 600);
+            _unit.LoadPolicy(TestBlockchain.GetStore().GetSnapshot());
+            AddTransaction(CreateTransaction(100000001));
+            AddTransaction(CreateTransaction(100000001));
+            AddTransaction(CreateTransaction(100000001));
+            AddTransaction(CreateTransaction(1));
+            _unit.VerifiedCount.Should().Be(4);
+            _unit.UnVerifiedCount.Should().Be(0);
+
+            _unit.InvalidateVerifiedTransactions();
+            _unit.VerifiedCount.Should().Be(0);
+            _unit.UnVerifiedCount.Should().Be(4);
+
+            AddTransactions(511); // Max per block currently is 512
+            _unit.VerifiedCount.Should().Be(511);
+            _unit.UnVerifiedCount.Should().Be(4);
+
+            var result = _unit.ReVerifyTopUnverifiedTransactionsIfNeeded(1, Blockchain.Singleton.GetSnapshot());
+            result.Should().BeTrue();
+            _unit.VerifiedCount.Should().Be(512);
+            _unit.UnVerifiedCount.Should().Be(3);
+
+            result = _unit.ReVerifyTopUnverifiedTransactionsIfNeeded(2, Blockchain.Singleton.GetSnapshot());
+            result.Should().BeTrue();
+            _unit.VerifiedCount.Should().Be(514);
+            _unit.UnVerifiedCount.Should().Be(1);
+
+            result = _unit.ReVerifyTopUnverifiedTransactionsIfNeeded(3, Blockchain.Singleton.GetSnapshot());
+            result.Should().BeFalse();
+            _unit.VerifiedCount.Should().Be(515);
+            _unit.UnVerifiedCount.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void TestTryAdd()
+        {
+            var tx1 = CreateTransaction();
+            _unit.TryAdd(tx1.Hash, tx1).Should().BeTrue();
+            _unit.TryAdd(tx1.Hash, tx1).Should().BeFalse();
+            _unit2.TryAdd(tx1.Hash, tx1).Should().BeFalse();
```

Comment:
Why is _unit2 returning `False`?

We should be able to add `tx1` to unit2, yes?

Reply:
@eryeer, if so, maybe let's also delete this test for now and set a remember on the issue #1101

Reply:
@vncoelho  _unit2 is a new mempool instance whose capcity is 0, so transaction cannot to be added into _uinit2. 
This test is used to cover line https://github.com/neo-project/neo/blob/e1fdde307153879f0a7ae64bd3e11d042284896e/neo/Ledger/MemoryPool.cs#L273-L274

Reply:
Makes sense, @eryeer, last time I reviewed I remembered that I considered that.
Thanks for highlighting again.